flytestdlib/resolver/k8s_resolver.go

```diff
@@ -0,0 +1,179 @@
+package resolver
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+
+	"google.golang.org/grpc/grpclog"
+	"google.golang.org/grpc/resolver"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/flyteorg/flyte/flytestdlib/logger"
+)
+
+const (
+	K8sSchema = "k8s"
+)
+
+type targetInfo struct {
+	serviceName      string
+	serviceNamespace string
+	port             string
+}
+
+func (t targetInfo) String() string {
+	return fmt.Sprintf("%s:///%s.%s:%s", K8sSchema, t.serviceNamespace, t.serviceName, t.port)
+}
+
+// NewBuilder creates a kubeBuilder which is used by grpc resolver.
+func NewBuilder(ctx context.Context, client kubernetes.Interface, schema string) resolver.Builder {
```

Comment:
If we don't intend to expose the scheme, maybe we could have another builder only for test.

Reply:
qq: can't we use `NewBuilder` to create a fake kube Builder?

Reply:
I meant we are exposing `schema string` unnecessarily to users because they are not supposed to freely choose any string they like.